guardian/mixins.py

```diff
@@ -310,11 +312,19 @@ def get_get_objects_for_user_kwargs(self, queryset: QuerySet) -> dict:
         """Get kwargs to pass to `get_objects_for_user`.
 
         Returns:
-            kwargs that should be passed to `get_objects_for_user`.
+            dict of kwargs to be passed to `get_objects_for_user`.
 
         Parameters:
             queryset (QuerySet): Queryset to filter.
+
+        Warning: Deprecation Warning
+            This method is deprecated and will be removed in future versions.
+            Use `get_user_object_kwargs` instead which has identical behavior.
         """
+        warnings.warn(
+            "get_get_objects_for_user_kwargs will be deprecated in a future release.",
+            DeprecationWarning,
```

Comment:
Should we use the new `@deprecated` decorator instead? It would require a dependency on `typing-extensions`, but it is the future of expressing deprecations.

- https://peps.python.org/pep-0702/#backwards-compatibility

Reply:
I didn't know the `@deprecated` decorator was back ported in `typing-extensions`. I'll take a look.